content-security-policy/script-src/script-nonces-hidden.html

```diff
@@ -0,0 +1,116 @@
+      assert_equals(document.querySelector('[nonce=""]'), script);
+      assert_equals(document.querySelector('[nonce=abc]'), null);
+
+      assert_equals(script.getAttribute('nonce'), '');
+      assert_equals(script.nonce, 'abc');
+    }, "HTML: Reading 'nonce' content attribute and IDL attribute.");
+
+    // Clone node.
+    test(t => {
+      script.setAttribute('executed', 'boo');
+      var s2 = script.cloneNode();
+      assert_equals(s2.nonce, 'abc', 'IDL attribute');
+      assert_equals(s2.getAttribute('nonce'), '');
+    }, "HTML: Cloned node retains nonce.");
+
+    async_test(t => {
+      var s2 = script.cloneNode();
+      document.head.appendChild(s2);
+      window.addEventListener('load', t.step_func_done(_ => {
+        assert_equals(s2.nonce, 'abc');
+        assert_equals(s2.getAttribute('nonce'), '');
+
+        // The cloned script won't execute, as its 'already started' flag is set.
+        assert_equals(s2.getAttribute('executed'), 'boo');
+      }));
+    }, "HTML: Cloned node retains nonce when inserted.");
+
+    // Set the content attribute to 'foo'
+    test(t => {
+      script.setAttribute('nonce', 'foo');
+      assert_equals(script.getAttribute('nonce'), 'foo');
+      assert_equals(script.nonce, 'abc');
+    }, "HTML: Writing 'nonce' content attribute.");
+
+    // Set the IDL attribute to 'bar'
+    test(t => {
+      script.nonce = 'bar';
+      assert_equals(script.nonce, 'bar');
+      assert_equals(script.getAttribute('nonce'), 'foo');
+    }, "HTML: Writing 'nonce' IDL attribute.");
+
+    // Fragment parser.
+    var documentWriteTest = async_test("Document-written script executes.");
+    document.write(`<script nonce='abc'>
+      documentWriteTest.done();
+      test(t => {
+        var script = document.currentScript;
+        assert_equals(script.getAttribute('nonce'), '');
+        assert_equals(script.nonce, 'abc');
+      }, "HTML: Document-written script's nonce value.");
+    </scr` + `ipt>`);
+
+    // Create node.
+    async_test(t => {
+      var s = document.createElement('script');
+      s.innerText = script.innerText;
+      s.nonce = 'abc';
+      document.head.appendChild(s);
+
+      window.addEventListener('load', t.step_func_done(_ => {
+        assert_equals(s.nonce, 'abc');
+        assert_equals(s.getAttribute('nonce'), null);
+        assert_equals(s.getAttribute('executed'), 'yay');
+      }));
+    }, "HTML: createElement.nonce.");
+
+    // Create node.
+    async_test(t => {
+      var s = document.createElement('script');
+      s.innerText = script.innerText;
+      s.setAttribute('nonce', 'abc');
+      assert_equals(s.getAttribute('nonce'), 'abc', "Pre-insertion content");
+      assert_equals(s.nonce, '', "Pre-insertion IDL");
+      document.head.appendChild(s);
+
+      window.addEventListener('load', t.step_func_done(_ => {
+        assert_equals(s.nonce, 'abc', "Post-insertion IDL");
+        assert_equals(s.getAttribute('nonce'), '', "Post-insertion content");
```

Comment:
I think the above two should be asserted directly after `document.head.appendChild(s);`